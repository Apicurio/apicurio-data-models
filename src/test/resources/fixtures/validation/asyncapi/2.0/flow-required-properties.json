{
  "asyncapi": "2.0.0",
  "info": {
      "title": "AsyncAPI Sample App",
      "license": {
      "name": "Apache 2.0",
      "url": "https://www.apache.org/licenses/LICENSE-2.0"
    },
     "version": "1.0.1"
  },
  "components": {
    "securitySchemes": {
     "supportedOauthFlows": {
        "type": "oauth2",
        "description": "Flows to support OAuth 2.0",
        "flows": {
          "implicit": {
          },
          "password": {
          },
          "clientCredentials": {
          },
          "authorizationCode": {
          }
        }
      }
    }
 }
}