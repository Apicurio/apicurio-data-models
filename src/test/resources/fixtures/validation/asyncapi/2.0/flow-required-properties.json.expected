[FLOW-006] |medium| {/components/securitySchemes[supportedOauthFlows]/flows/implicit->scopes} :: OAuth Flow is missing defined scopes.
[FLOW-001] |medium| {/components/securitySchemes[supportedOauthFlows]/flows/implicit->authorizationUrl} :: Implicit OAuth Flow is missing an Authorization URL.
[FLOW-006] |medium| {/components/securitySchemes[supportedOauthFlows]/flows/password->scopes} :: OAuth Flow is missing defined scopes.
[FLOW-002] |medium| {/components/securitySchemes[supportedOauthFlows]/flows/password->tokenUrl} :: Password OAuth Flow is missing a Token URL.
[FLOW-006] |medium| {/components/securitySchemes[supportedOauthFlows]/flows/clientCredentials->scopes} :: OAuth Flow is missing defined scopes.
[FLOW-002] |medium| {/components/securitySchemes[supportedOauthFlows]/flows/clientCredentials->tokenUrl} :: Client Credentials OAuth Flow is missing a Token URL.
[FLOW-006] |medium| {/components/securitySchemes[supportedOauthFlows]/flows/authorizationCode->scopes} :: OAuth Flow is missing defined scopes.
[FLOW-001] |medium| {/components/securitySchemes[supportedOauthFlows]/flows/authorizationCode->authorizationUrl} :: Auth Code OAuth Flow is missing an Authorization URL.
[FLOW-002] |medium| {/components/securitySchemes[supportedOauthFlows]/flows/authorizationCode->tokenUrl} :: Auth Code OAuth Flow is missing a Token URL.